src/Sylius/Bundle/CoreBundle/Telemetry/Serializer/TelemetryDataNormalizer.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Telemetry\Serializer;
+
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Business\CurrenciesData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Business\LocalesData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Business\MetricsCountsData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Business\OrderMetricsData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Business\OrdersBusinessData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Business\PaymentMethodsData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Business\PaymentProviderData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Business\ShippingMethodsData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Business\ShippingProviderData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Plugins\InstalledPluginsData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Plugins\PluginData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Technical\DatabaseData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Technical\EnvironmentData;
+use Sylius\Bundle\CoreBundle\Telemetry\DTO\Technical\VersionData;
+use Sylius\Component\Core\Telemetry\DTO\TelemetryDataInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/** @internal */
+final class TelemetryDataNormalizer implements NormalizerInterface
+{
+    public function normalize(mixed $object, ?string $format = null, array $context = []): array
+    {
+        return match (true) {
```

Comment:
I feel that it would be better if these DTO classes themselves were responsible for proper normalization to the array